### PR TITLE
Add Model Functionality

### PR DESCRIPTION
**Pull request by lpile**
Add Model Functionality
--
* Add User model validations spec
* Add User model
* Add Item model validations spec
* Add Item model
* Add Order model validations spec
* Add Order model
* Add Order_item model validations spec
* Add OrderItem model
* Add relationships spec for all models
* Add relationships to all models
* All tests passing

### CODE REVIEW

File: app/models/order_item.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class OrderItem < ApplicationRecord
+  validates_presence_of :quantity, :price, :fulfilled
+
```
Comment (DavisC0801) [inline app/models/order_item.rb]: We're also able to add validations for numericality for the integers and floats, I prefer to double-validate when possible.

Comment (DavisC0801) [inline app/models/order_item.rb]:
```suggestion
validates :quantity, numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than_or_equal_to: 10000}
```

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  validates :email, uniqueness: true, presence: true
+  validates_presence_of :password, :role, :active, :address, :city, :state, :zip, require: true
+
```
Comment (DavisC0801) [inline app/models/user.rb]:
```suggestion
it {should validate_numericality_of(:role).only_integer}
```

As above, we can include tests for the numericality validations in the models. This is for pretty much all numerical inputs but I don't want to put that in every place.

File: app/models/order_item.rb (hunk 1 above)
Comment (lpile) [reply app/models/order_item.rb]: I agree, over-validating makes the model more solid.

File: app/models/user.rb (hunk 2 above)
Comment (lpile) [reply app/models/user.rb]: Cool, I didn't know this.

Comment (lpile) [reply app/models/user.rb]: I moved suggest code to user_spec

File: app/models/order_item.rb (hunk 1 above)
Comment (lpile) [reply app/models/order_item.rb]: I've add suggest code to model

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  validates :email, uniqueness: true, presence: true
+  validates_presence_of :password, :role, :active, :address, :city, :state, :zip, require: true
+  validates_numericality_of :role, only_integer: true
```
Comment (DavisC0801) [inline app/models/user.rb]: This is good for now, just make a mental note to clamp these once we define what we need, for example if our roles are 0-4 then we'll make a validation to make sure we only have that as inputs.

**Pull request merged**